libs/@guardian/libs/src/measure/startMeasure.ts

```diff
@@ -31,16 +34,28 @@ export const startMeasure = (
 		start: performance.now(),
 	} satisfies PerformanceMeasureOptions;
 
-	const endMeasure = () => {
-		const measureName = serialise(team, name, action);
+	const measureName = serialise(team, name, action);
 
-		const { duration } = window.performance.measure(measureName, options) ??
+	const endMeasure = (end?: number) => {
+		const { duration } = window.performance.measure(measureName, {
+			...options,
+			end,
+		}) ??
 			window.performance.getEntriesByName(measureName, 'measure').at(-1) ?? {
 				duration: performance.now() - options.start,
 			};
 
 		return Math.ceil(duration);
 	};
 
-	return { endMeasure };
+	const observe = () =>
+		new PerformanceObserver((entries, observer) => {
+			const [endMark] = entries.getEntriesByName(measureName);
```

Comment:
This is cool 😎 . Just a suggestion re semantics and dev tools. Using the zeroth mark as the semantic end mark is a little unusual. I'm wondering if it is less surprising to anyone looking at the performance dev tools to always create start marks. Then we can grab the [1]'st mark as the end mark for `observe`.